instrumentation/opentelemetry-instrumentation-sqlalchemy/src/opentelemetry/instrumentation/sqlalchemy/__init__.py

```diff
@@ -158,17 +160,22 @@ def _instrument(self, **kwargs):
                 "create_async_engine",
                 _wrap_create_async_engine(tracer_provider, enable_commenter),
             )
+
+        self.engines = []
```

Comment:
Is there any reason to keep list of engines, why is it necessary to keep all the engines in array?

Reply:
You can create one engine or engines, so if I keep them in array it will be easier to clean up

Reply:
No problem, We should consider when we will remove the engine from that list as well, otherwise potentially a memory leak could happen

Reply:
No problem, We should consider when we will remove the engine from that list as well, otherwise potentially a memory leak could happen

Reply:
i wonder if we should implement call uninstrument logic when the garbage collector trying to collect the engine 
https://docs.python.org/3.6/library/weakref.html#finalizer-objects

Reply:
I thought about this but the problem is after the instrumentation the EngineTracer returns to the user and we don't know when the garbage collector works

Reply:
Thank you for the comments